util/src/test/java/io/grpc/util/OutlierDetectionLoadBalancerTest.java

```diff
@@ -355,6 +365,38 @@ public void delegatePick() throws Exception {
         readySubchannel);
   }
 
+  /**
+   * Any ClientStreamTracer.Factory set by the delegate picker should still get used.
+   */
+  @Test
```

Comment:
Do you also want to add a test-case when the `delegateFactory` is `null` in the `ResultCountingClientStreamTracerFactory(AddressTracker tracker, ClientStreamTracer.Factory delegateFactory)` constructor? Just to make sure there are no NPE possibilities? So this would require `mockStreamTracerFactory` to be set to `null` in the new test-case.

Reply:
Good idea, I added that test.